CMakeLists.txt

```diff
@@ -235,6 +235,18 @@ if ( ZEEK_SANITIZERS )
 
     set(_sanitizer_flags "-fsanitize=${ZEEK_SANITIZERS}")
 
+    # The linker command used by check_cxx_compiler_flag requires you to also pass the sanitizer to
+    # it or it fails. The best way to do this is to set CMAKE_REQUIRED_LINK_OPTIONS, but save off a
+    # copy of it so it can be reset back to what it was previously afterwards.
+    set(_temp_link_options ${CMAKE_REQUIRED_LINK_OPTIONS})
+    set(CMAKE_REQUIRED_LINK_OPTIONS ${_sanitizer_flags})
```

Comment:
I'm wondering if we should append this to the current value, in case that comes with other options needed?

Reply:
I'm trying to think whether that variable would be set at all before CMake gets to this point, and I don't think it would be, but you're probably right that it's best to assume it would.